### PR TITLE
Add copy all options to register script

### PR DESCRIPTION
**Pull request by bgedik**
## Why are the changes needed?

`FlyteRemote.register_script()` copies the entire source directory with some default ignore filters, but does not let the user customize the ignore filters.

## What changes were proposed in this pull request?

API changes to allow additional ignore filters during workflow registery.

## How was this patch tested?

Unit tests for `fast_registration.py`

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [ ] All commits are signed-off.

## Related PRs

?

## Docs link

?

### CODE REVIEW

File: flytekit/tools/fast_registration.py [hunk 1]
```diff
@@ -8,20 +8,36 @@
...
 
 
-def fast_package(source: os.PathLike, output_dir: os.PathLike, deref_symlinks: bool = False) -> os.PathLike:
+@dataclass(frozen=True)
+class FastPackageOptions:
+    """
+    FastPackageOptions is used to set configuration options when packaging files.
+    """
+
+    ignores: list[Ignore] = None
```
Comment (wild-endeavor) [inline flytekit/tools/fast_registration.py:31]:
```suggestion
    ignores: Optional[list[Ignore]] = None
```

File: flytekit/remote/remote.py [hunk 2]
```diff
@@ -965,7 +970,7 @@ def register_script(
         project: typing.Optional[str] = None,
         domain: typing.Optional[str] = None,
         destination_dir: str = ".",
-        copy_all: bool = False,
+        copy_all: typing.Union[bool, FastPackageOptions] = False,
```
Comment (wild-endeavor) [inline flytekit/remote/remote.py:973]: can we make this a separate arg?  it can be ignored if copy is not used.

**Comment by wild-endeavor** [comment]
thanks @bgedik - the copy_all union though, could we split that out please?  I think it's nicer to have that explicitly split out.

**Review by wild-endeavor — APPROVED** [review]

**Comment by wild-endeavor** [comment]
test error looks like it's not related, but take a look at lint?

**Comment by bgedik** [comment]
> test error looks like it's not related, but take a look at lint?

@wild-endeavor  The lint is fixed, but there is now an unrelated error from Mac build

**Comment by wild-endeavor** [comment]
@bgedik not sure what is happening with the test failure, but it's been fixed on master.  Can you pull in master and push to this pr please?

**Pull request merged**